torch/python/src/sphericart_torch/__init__.py

```diff
@@ -117,4 +125,70 @@ def compute(
             return self._sph.compute(xyz), None
 
 
-__all__ = ["SphericalHarmonics"]
+def e3nn_spherical_harmonics(
+    l_list: Union[List[int], int],
+    x: torch.Tensor,
+    normalize: Optional[bool] = False,
+    normalization: Optional[str] = "integral",
+) -> torch.Tensor:
+    """
+    Computes spherical harmonics with an interface similar to the e3nn package.
+
+    Provides an interface that is similar to :py:func:`e3nn.o3.spherical_harmonics`
+    but uses :py:class:`SphericalHarmonics` for the actual calculation.
+    Uses the same ordering of the [x,y,z] axes, and supports the same options for
+    input and harmonics normalization as :py:mod:`e3nn`. However, it does not support
+    defining the irreps through a :py:class:`e3nn.o3._irreps.Irreps` or a string
+    specification, but just as a single integer or a list of integers.
+
+    :param l_list:
+        Either a single integer or a list of integers specifying which
+        :math:`Y^l_m` should be computed. All values up to the maximum
+        l value are computed, so this may be inefficient for use cases
+        requiring a single, or few, angular momentum channels.
+    :param x:
+        A `torch.Tensor` containing the coordinates, in the same format
+        expected by the `e3nn` function.
+    :param normalize:
+        Flag specifying whether the input positions should be normalized,
+        or whether the function should compute scaled :math:`\tilde{Y}^l_m`
+    :param normalization:
+        String that can be "integral", "norm", "component", that controls
+        a further scaling of the :math:`Y_m^l`. See the
+        documentation of :py:func:`e3nn.o3.spherical_harmonics()`
+        for a detailed explanation of the different conventions.
+    """
+
+    if not hasattr(l_list, "__len__"):
+        l_list = [l_list]
+    l_max = max(l_list)
+    sh = SphericalHarmonics(l_max, normalized=normalize).compute(x[:, [2, 0, 1]])[0]
+
+    if normalization != "integral":
+        sh *= math.sqrt(4 * math.pi)
+
+    sh_list = []
```

Comment:
Is there any performance difference with a list + cat vs torch.empty + assign?

Reply:
ok so I implemented the "do nothing if you don't need to do something" approach. should be good enough at least until we don't get specific requests from e3nn people.